versioned/tiered/src/main/java/com/dremio/nessie/tiered/builder/L1Builder.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.tiered.builder;
+
+import com.dremio.nessie.versioned.Key;
+import com.dremio.nessie.versioned.store.Id;
+
+/**
+ * Interface to create an L1 Builder. To be implemented by each {@link Store} implementation.
+ *
+ * @param <T> The concrete type built by this builder.
+ */
+public interface L1Builder<T> {
+
+  /**
+   * Construct the expected object.
+   * @return The built object.
+   */
+  T build();
```

Comment:
Why is it parameterized? It looks like an implementation detail

Reply:
Think it's okay to have it parameterized.
If you use a `L1.builder()` you get a `L1Builder<L1>`, which builds an `L1` object.
Similar from a store's point-of-view, just that only the store has the type-safety, which is probably fine.